src/transformers/generation_utils.py

```diff
@@ -711,14 +1044,28 @@ def greedy_search(
                 The id of the `padding` token.
             eos_token_id (:obj:`int`, `optional`):
                 The id of the `end-of-sequence` token.
+            output_attentions (:obj:`bool`, `optional`, defaults to `False`):
+                Whether to return attention weights in output
+            output_hidden_states (:obj:`bool`, `optional`, defaults to `False`):
+                Whether to return hidden states in output
+            output_scores (:obj:`bool`, `optional`, defaults to `False`):
+                Whether to return the scores in output
+            return_dict_in_generate (:obj:`bool`, `optional`, defaults to `False`):
+                Whether the output should be a dict (:obj:`GreedySearchOutput`) or just the generated sequence
+                (:obj:`torch.LongTensor`).
+
             model_kwargs:
                 Additional model specific keyword arguments will be forwarded to the :obj:`forward` function of the
                 model. If model is an encoder-decoder model the kwargs should include :obj:`encoder_outputs`.
 
+
         Return:
-            :obj:`torch.LongTensor` of shape :obj:`(batch_size * num_return_sequences, sequence_length)`: The generated
-            sequences. The second dimension (sequence_length) is either equal to :obj:`max_length` or shorter if all
-            batches finished early due to the :obj:`eos_token_id`.
+            Either a :obj:`torch.LongTensor` object (default behaviour) or a
+            :class:`~transformers.generation_utils.GreedySearchDecoderOnlyOutput` if
+            ``model.config.is_encoder_decoder=False`` and ``return_dict_in_generate=True`` or a
+            :class:`~transformers.generation_utils.GreedySearchEncoderDecoderOutput` if
+            ``model.config.is_encoder_decoder=True``.
```

Comment:
This needs to be of the format `type: explanation` for proper rendering in the docs.
```suggestion
            :class:`~transformers.generation_utils.GreedySearchDecoderOnlyOutput`, :class:`~transformers.generation_utils.GreedySearchEncoderDecoderOutput` or obj:`torch.LongTensor`: A :obj:`torch.LongTensor` containing xxx (default behaviour) or a :class:`~transformers.generation_utils.GreedySearchDecoderOnlyOutput` if
            ``model.config.is_encoder_decoder=False`` and ``return_dict_in_generate=True`` or a
            :class:`~transformers.generation_utils.GreedySearchEncoderDecoderOutput` if
            ``model.config.is_encoder_decoder=True``.
```